### DISCUSSION TITLE
Is custom GQL Query/Mutation logic possible?

### DISCUSSION THREAD

**Discussion by ConcatenatedNonsense** [Azure/data-api-builder#1916] [Q&A]
DAB adds some lovely CRUD functions for entities by default but I'm looking to extend that and want to know if it's possible.

Under default behaviour DAB is generating a Query group with:
- Get a list of all
- Get by PK

and a Mutation group with:
- Create
- Update
- Delete

Am I able to do either or both of the following:
- Define more complex queries which are then added to the existing Query/Mutation groups?
- Define a schema for a new group with custom GQL queries which is then autogenerated by DAB?

**Comment by seantleonard** [comment]
Hi @ConcatenatedNonsense, we don't have the two items you mention at the moment.

Can you elaborate more and/or provide an example for your ask: "Define a schema for a new group with custom GQL queries which is then autogenerated by DAB?"
